Python/specialize.c

```diff
@@ -814,15 +819,47 @@ _Py_Specialize_LoadMethod(PyObject *owner, _Py_CODEUNIT *instr, PyObject *name,
     cache1->tp_version = owner_cls->tp_version_tag;
 
     if (kind != METHOD || descr == NULL) {
-#if SPECIALIZATION_STATS
         if (kind == GETSET_OVERRIDDEN &&
             PyObject_TypeCheck(owner, &PyBaseObject_Type)) {
```

Comment:
Move this check to before the `analyze_descriptor`. When accessing an attribute of a class, `LOAD_METHOD` is basically the same as `LOAD_ATTR`. The only difference is the NULL left on the stack at the end.